File: app/src/main/java/us/martypants/rightpointdemo/network/DataManagerAPI.java
package us.martypants.rightpointdemo.network;

import retrofit2.http.GET;
import retrofit2.http.Query;
import rx.Observable;
import us.martypants.rightpointdemo.models.ImdbData;

/**
 * Created by marty on 7/28/16.
 */


public interface DataManagerAPI {


    @GET("/?apikey=2ec4b519")
    Observable<ImdbData> getImdbData(@Query("s") String search, @Query("page") int page);

    @GET("/?apikey=2ec4b519")
    Observable<ImdbData> getSpecificType(@Query("s") String search, @Query("type") String type, @Query("page") int page);

}

File: app/src/main/java/us/martypants/hca/network/DataManagerAPI.kt
package us.martypants.hca.network

import retrofit2.http.GET
import retrofit2.http.Query
import rx.Observable
import us.martypants.hca.models.SoData

interface DataManagerAPI {

    @GET("/2.2/questions?order=desc&sort=activity&site=stackoverflow&pagesize=100")
    fun getSoData(@Query("page") page: Int): Observable<SoData>?
}